### PR TITLE
Upgrade Gradle setup

### PR DESCRIPTION
**Pull request by yuri-sergiichuk**
In this PR I have upgraded the Gradle setup of the repository as well as introduced usages of the latest Gradle Kotlin best practices.
As part of the PR, I have also updated dependencies and resolved code quality tooling warnings.

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -43,5 +43,7 @@ allprojects {
 
 subprojects {
     apply<JavaConventionPlugin>()
+    apply<JavadocConventionPlugin>()
     apply<DependencyManagementPlugin>()
+    apply<CodeQualityPlugin>()
```
Comment (alexander-yevsyukov) [inline build.gradle.kts:48]: This is really cool!

File: gradle/wrapper/gradle-wrapper.properties [hunk 2]
```diff
@@ -1,5 +1,5 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-6.8.2-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-6.9.1-bin.zip
```
Comment (alexander-yevsyukov) [inline gradle/wrapper/gradle-wrapper.properties:3]: Did you try 7.2?

File: cloudbuild.yaml [hunk 3]
```diff
@@ -28,7 +28,7 @@ steps:
   # The following step starts the build process using Gradle official image, performs the build
   # and runs `jib` for the specified `gcpProject` in order to build and deploy the container
   # image to the Google Container Registry.
-  - name: 'gradle:6.8-jdk11'
+  - name: 'gradle:6.9-jdk11'
```
Comment (alexander-yevsyukov) [inline cloudbuild.yaml:31]: Did you try 7.2?

**Review by alexander-yevsyukov — APPROVED** [review]
LGTM, suggesting to migrate to Gradle 7.2 in this PR or soon after.

File: gradle/wrapper/gradle-wrapper.properties (hunk 2 above)
Comment (yuri-sergiichuk) [reply gradle/wrapper/gradle-wrapper.properties:3]: Yep. Spine 1.x (bootstrap plugin) is not compatible with Gradle 7.x. I plan to migrate the app to Spine 2.x and do the migration to Gradle 7.x there.

File: cloudbuild.yaml (hunk 3 above)
Comment (yuri-sergiichuk) [reply cloudbuild.yaml:31]: Yep. Spine 1.x (bootstrap plugin) is not compatible with Gradle 7.x. I plan to migrate the app to Spine 2.x and do the migration to Gradle 7.x there.

**Pull request merged**